cumulus/pallets/collator-selection/src/lib.rs

```diff
@@ -424,8 +460,11 @@ pub mod pallet {
 			let who = ensure_signed(origin)?;
 
 			// ensure we are below limit.
-			let length = <Candidates<T>>::decode_len().unwrap_or_default();
-			ensure!((length as u32) < Self::desired_candidates(), Error::<T>::TooManyCandidates);
+			let length: u32 = <CandidateList<T>>::decode_len()
+				.unwrap_or_default()
+				.try_into()
+				.unwrap_or_default();
```

Comment:
It's ok here since `CandidateList` has `ValueQuery`, which means if there is nothing in storage a `get` will return an empty list but `decode_len` would return `None`, which would actually mean `0` length for us, which also means the `try_into` would be infallible. If, on the other hand, the value couldn't decode, it would just fail later in the function with an error when trying to mutate the list.